src/notifications/types.ts

```diff
@@ -55,3 +56,7 @@ export type NotificationSourceRequestContext = NotificationBaseContext & {
 export type NotificationDoneByContext = NotificationBaseContext & {
   doneBy: Reference<User>;
 };
+
+export type NotificationRolesContext = NotificationSourceContext & {
```

Comment:
Again to make it a bit more explicit. I'm afraid `Role` is too broad term

```suggestion
export type NotificationSourceMemberRoleContext = NotificationSourceContext & {
```